File: java-dsa/src/algorithms/InsertionSort.java
package algorithms;

import java.util.Arrays;

public class InsertionSort {
    public static void main(String[] args) {
        int[] arr = {13, 46, 24, 52, 20, 9};
        int n = arr.length;

        for (int i = 0; i < n - 1; i++) {
            int j = i + 1;
            while (j > 0 && arr[j - 1] > arr[j]) {
                int temp = arr[j - 1];
                arr[j - 1] = arr[j];
                arr[j] = temp;
                j--;
            }
        }

        System.out.println(Arrays.toString(arr));
    }
}

File: python/algorithms/insertion_sort.py
arr = [13, 46, 24, 52, 20, 9]
n = len(arr)

for i in range(0, n - 1):
    j = i + 1
    while j > 0 and arr[j - 1] > arr[j]:
        arr[j], arr[j - 1] = arr[j - 1], arr[j]
        j = j - 1

print(arr)
